app/assets/javascripts/admin/views/auth/login_view.js

```diff
@@ -24,9 +25,15 @@ class LoginView extends React.PureComponent<Props> {
     this.props.form.validateFields(async (err: ?Object, formValues: Object) => {
       if (!err) {
         await Request.sendJSONReceiveJSON("/api/auth/login", { data: formValues });
-        const user = await getActiveUser();
-        Store.dispatch(setActiveUserAction(user));
-        this.props.history.push("/dashboard");
+        if (!Utils.hasUrlParam("redirectPage")) {
+          const user = await getActiveUser();
+          Store.dispatch(setActiveUserAction(user));
+          this.props.history.push("/dashboard");
+        } else {
+          const user = await getActiveUser();
+          Store.dispatch(setActiveUserAction(user));
+          window.location.replace(Utils.getUrlParamValue("redirectPage"));
+        }
```

Comment:
In the case were we only want to redirect there is no need to request a user. Hence, delete lines 33 & 34.